src/gsy_e/gsy_e_core/simulation.py

```diff
@@ -244,28 +244,34 @@ def update_area_stats(self, area, endpoint_buffer):
             endpoint_buffer.results_handler.all_ui_results["kpi"].get(area.uuid, {}))
 
     def _update_and_send_results(self):
-        self.endpoint_buffer.update_stats(
-            self.area, self.status, self.progress_info, self.current_state)
-
-        self.update_area_stats(self.area, self.endpoint_buffer)
-
-        if self.export_results_on_finish:
-            if (self.area.current_market is not None
-                    and gsy_e.constants.RETAIN_PAST_MARKET_STRATEGIES_STATE):
-                # for integration tests:
-                self.export.raw_data_to_json(
-                    self.area.current_market.time_slot_str,
-                    self.endpoint_buffer.flattened_area_core_stats_dict
-                )
-
-            self.file_stats_endpoint(self.area)
-
-        elif self.should_send_results_to_broker:
+        if self.should_send_results_to_broker:
```

Comment:
Minor, but I think the change in the order will change the meaning of this if / elif block. To be more precise, the former behavior stated that "if there is no configuration to expose the simulation results at the end of the simulation, then check if the results should be sent to the broker", and the updated order is exactly the opposite. This will affect the scenario where the broker is online (therefore the should_send_results_to_broker variable will be true) and we want to export the results. With the previous implementation this worked without a problem, because the export check took precedence over the broker check, which is not the case now.

Regardless of the above, to my eyes it is optional since a CLI user will probably not have Kafka installed and running anyways.

Reply:
Ah yes, let me try to also take this into account.

Reply:
@spyrostz 
I gave it another look and I think there is no need for a change, as the case you are mentioning :
> where the broker is online (therefore the should_send_results_to_broker variable will be true) and we want to export the results

We only want to export the results when we start the gsy-e from the cli, right? The `should_send_results_to_broker` already contains a check if the gsy-e is started from cli: https://github.com/gridsingularity/gsy-e/blob/2984dc21b1ddb6330e031617f6e676b649925265/src/gsy_e/gsy_e_core/simulation.py#L437

So, AFAIU, the first if block will not be triggered when we run the gsy-e locally but the Kafka connection is enabled.
What do you think?

Reply:
You are right, I did not recall that we had already adapted it in order to explicitly define that the simulation was started from the CLI! Thanks for the clarification, it is good as it is!